cogs/birthday/birthday.py

```diff
@@ -512,52 +543,100 @@ async def setSelfBirthday(self, ctx: Context, *, birthday: MonthDayConverter):
         birthdayStr = "{0:%B} {0:%d}".format(birthday)
 
         if birthdayConfig:
-            confirmationStr = (
-                f"Are you sure you want to set your birthday to "
-                f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}? "
-                "Only administrators and moderators can reset your birthday afterwards. "
-                f"Type {bold('`yes`', escape_formatting=False)} to confirm."
+            confirmationStr = "\n".join(
+                (
+                    f"Are you sure you want to set your birthday to "
+                    f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}? "
+                    "Only administrators and moderators can reset your birthday afterwards.",
+                    f"Type {bold('`yes`', escape_formatting=False)} to confirm.",
+                )
             )
 
-            await ctx.author.send(f"{headerWarn}: {confirmationStr}")
+            async def mainFlow(destination, carefree: bool = False):
+                # define the time for the message to live for
+                SENSITIVE_MSG_TTL = None if carefree else 10.0
+                try:
+                    await destination.send(
+                        f"{headerWarn}: {confirmationStr}",
+                        delete_after=SENSITIVE_MSG_TTL,
+                    )
+                except discord.Forbidden:
+                    # catch failed DM
```

Comment:
nit: reword so that we don't mean this *only* occurs in DM (even though we expect the other case shouldn't not happen ever). Also add a note to say that this explicit re-`raise` is just for noting, 'cause otherwise without the `try` block, exception would occur anyway.

```suggestion
                    # this means messages cannot be sent to destination, 
                    # so we abort the flow now by re-raising the exception
                    # and the caller shall handle it gracefully
```

Reply:
Fixed